app/controllers/items_controller.rb

```diff
@@ -0,0 +1,35 @@
+class ItemsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_wishlist!, only: %i(new create)
+
+  def new
+    @item = @wishlist.items.new
+    authorize! :new, @item
+  rescue CanCan::AccessDenied
+    redirect_to @wishlist, alert: t("generic_error")
+  end
+
+  def create
+    @item = @wishlist.items.create(item_params)
+    authorize! :create, @item
+    @item.save!
+    redirect_to @wishlist, notice: t(".success")
+  rescue CanCan::AccessDenied
+    redirect_to @wishlist, alert: t("generic_error")
+  rescue ActiveRecord::RecordInvalid
+    render :new
+  end
+
+  private
+
+  def item_params
+    params[:item] ||= {}
+    params.require(:item).permit(:wishlist_id, :name, :description)
+  end
+
+  def set_wishlist!
```

Comment:
I wrote it the same way as in WishlistController which is already on master - I think it's good practise, because we would know if method throws an error, it wouldn't just fail silently :)